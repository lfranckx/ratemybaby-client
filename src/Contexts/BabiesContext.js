import React, { Component } from 'react';

const BabiesContext = React.createContext({
    usersBabies: [],
    babies: [],
    error: null,
    setError: () => {},
    clearError: () => {},
    setBabies: () => {},
    setUsersBabies: () => {}
})

export default BabiesContext

export class BabiesProvider extends Component {
    state = {
        usersBabies: [],
        babies: [],
        error: null
    }
    
    setBabies = babies => {
        this.setState({ babies })
    }

    setUsersBabies = babies => {
        console.log('setUsersBabies:', babies);
        this.setState({ usersBabies: babies })
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    render() {
        const value = {
            babies: this.state.babies,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setBabies: this.setBabies,
            setUsersBabies: this.setUsersBabies
        }
        return (
            <BabiesContext.Provider value={value}>
                {this.props.children}
            </BabiesContext.Provider>
        )
    }
}