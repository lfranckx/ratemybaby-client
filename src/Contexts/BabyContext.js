import React, { Component } from 'react'
import BabyApiService from '../Services/baby-api-service'

export const nullBaby = {
    baby: null
}

const BabyContext = React.createContext({
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    updateBaby: () => {}
})

export default BabyContext

export class BabyProvider extends Component {
    state = {
        baby: nullBaby,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        console.log('running BabyContext clearError()');
        this.setState({ error: null })
    }

    setBaby = baby => {
        console.log('setting baby:', baby);
        this.setState({ 
            baby: baby 
        })
    }

    clearBaby = () => {
        console.log('running BabyContext clearBaby()');
        this.setBaby(nullBaby)
    }   

    updateBaby = baby => {
        console.log('updating baby:', baby, 
        'total_score:', baby.total_score,
        'total_votes:', baby.total_votes)
        this.setState({ 
            ...this.state.baby,
            baby 
        })
        BabyApiService.updateBaby(baby)
    }
      
    render() {
        console.log('BabyContext State', this.state);
        
        const value = {
            baby: this.state.baby,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setBaby: this.setBaby,
            clearBaby: this.clearBaby,
            updateBaby: this.updateBaby,
            clearUser: this.clearUser
        }
        return (
            <BabyContext.Provider value={value}>
                {this.props.children}
            </BabyContext.Provider>
        )
    }
}