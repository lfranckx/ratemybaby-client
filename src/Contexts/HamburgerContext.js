import React, { Component } from 'react'

export const nullBaby = {
    baby: null
}

const HamburgerContext = React.createContext({
    active: false,
    usersBabies: [],
    babies: [],
    baby: nullBaby,
    error: null,
    setError: () => {},
    clearError: () => {},
    toggleActive: () => {},
    setNotActive: () => {},
    setBaby: () => {},
    clearBaby: () => {},
    setUsersBabies: () => {}
})

export default HamburgerContext

export class HamburgerProvider extends Component {
    state = {
        active: false,
        usersBabies: [],
        babies: [],
        baby: nullBaby,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    toggleActive = () => {
        this.setState({ active: !this.state.active })
    }

    setNotActive = () => {
        this.setState({ active: false })
    }

    setBaby = baby => {
        this.setState({ 
            baby: baby 
        })
    }

    clearBaby = () => {
        this.setBaby(nullBaby)
    }   

    setUsersBabies = usersBabies => {
        this.setState({ usersBabies })
    }
      
    render() {
        const value = {
            active: this.state.active,
            usersBabies: this.state.usersBabies,
            babies: this.state.babies,
            baby: this.state.baby,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            toggleActive: this.toggleActive,
            setNotActive: this.setNotActive,
            setBabies: this.setBabies,
            setUsersBabies: this.setUsersBabies,
            setBaby: this.setBaby,
            clearBaby: this.clearBaby,
        }
        return (
            <HamburgerContext.Provider value={value}>
                {this.props.children}
            </HamburgerContext.Provider>
        )
    }
}