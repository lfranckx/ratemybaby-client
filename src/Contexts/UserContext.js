import React, { Component } from 'react'
// import UserApiService from '../Services/user-api-service'
import BabyApiService from '../Services/baby-api-service'

export const nullUser = {
    user: {}
}

const UserContext = React.createContext({
    user: nullUser,
    error: null,
    setError:() => {},
    clearError: () => {},
    setUser: () => {},
    clearUser: () => {},
    updateBaby: () => {},
    
})

export default UserContext

export class UserProvider extends Component {
    state = {
        user: nullUser,
        error: null
    }

    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setUser = user => {
        console.log('setting state from UserContext:', user);
        this.setState({ user })
    }

    clearUser = () => {
        this.setUser(nullUser)
    }   

    updateBaby = baby => {
        console.log('updating baby...', baby, 
        'total_score:', baby.total_score,
        'total_votes:', baby.total_votes)
        this.setState({ 
            ...this.state.baby,
            baby 
        })
        BabyApiService.updateBaby(baby)
    }

    render() {
        const value = {
            user: this.state.user,
            error: this.state.error,
            setError: this.setError,
            clearError: this.clearError,
            setUser: this.setUser,
            clearUser: this.clearUser,
            updateBaby: this.updateBaby
        }
        return (
            <UserContext.Provider value={value}>
                {this.props.children}
            </UserContext.Provider>
        )
    }
}