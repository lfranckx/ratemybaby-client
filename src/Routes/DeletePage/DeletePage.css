#delete-section {
    background-color: white;
    padding: 20px;
}

/* #delete-section h3 {

} */