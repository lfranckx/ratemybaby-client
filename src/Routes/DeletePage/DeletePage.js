import React, { Component } from 'react'
import BabyApiService from '../../Services/baby-api-service'
import BabyContext from '../../Contexts/BabyContext'
import { Link } from 'react-router-dom'
import './DeletePage.css'


export default class DeletePage extends Component {

    static contextType = BabyContext

    componentDidMount() {
        this.context.clearError()
        this.context.setNotActive()
        BabyApiService.getByParentId()
            .then(res => {
                this.context.setUsersBabies(res)
            })
            .catch(this.context.setError)
    }

    componentWillUnmount() {
        this.context.clearError()
        this.context.clearBaby()
        BabyApiService.getByParentId()
            .then(res => {
                this.context.setUsersBabies(res)
            })
            .catch(this.context.setError)
    }

    deleteBaby = (babyId) => {
        BabyApiService.deletBaby(babyId)
    }

    render() {
        const { baby } = this.context
        
        if (!baby) {
            return <div className='loading'>Loading...</div>
        }
        return ( 
            <section id="delete-section">
                <h3 >Are you sure you want to delete?</h3>
                <button 
                    id='delete-button'
                    onClick={() => {
                        this.deleteBaby(baby.id)
                    }}
                >
                    <Link 
                        id='delete-link' 
                        to='/rate'>
                            Delete
                    </Link>
                </button>
            </section>
            
        )
    }
}