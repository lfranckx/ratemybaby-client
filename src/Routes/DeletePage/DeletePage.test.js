import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DeletePage from './DeletePage';
import { HamburgerProvider } from '../../Contexts/HamburgerContext';

describe(`CreateProfileForm component`, () => {
    
    it(`renders loading without props`, () => {
        const wrapper = shallow(
            <HamburgerProvider>
                <DeletePage />
            </HamburgerProvider>
        )
        expect(toJson(wrapper)).toMatchSnapshot()
    });

})