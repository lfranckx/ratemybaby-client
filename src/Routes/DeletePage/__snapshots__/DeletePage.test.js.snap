// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfileForm component renders loading without props 1`] = `
<ContextProvider
  value={
    Object {
      "active": false,
      "babies": Array [],
      "baby": Object {
        "baby": null,
      },
      "clearBaby": [Function],
      "clearError": [Function],
      "error": null,
      "setBabies": undefined,
      "setBaby": [Function],
      "setError": [Function],
      "setNotActive": [Function],
      "setUsersBabies": [Function],
      "toggleActive": [Function],
      "usersBabies": Array [],
    }
  }
>
  <DeletePage
    history={
      Object {
        "push": [Function],
      }
    }
    location={Object {}}
    match={
      Object {
        "params": Object {},
      }
    }
  />
</ContextProvider>
`;
